server/src/controllers/christmas-tree.es6

```diff
@@ -206,15 +159,174 @@ const permitResult = (permit, svgData) => {
     }
   };
   return result;
 };
 
+const updatePermitWithToken = (res, permit, token) => {
+  const tcsAppID = vcapConstants.payGovAppId;
+  permit
+    .update({
+      paygovToken: token
+    })
+    .then(savedPermit => {
+      return res.status(200).json({
+        token: token,
+        permitId: savedPermit.permitId,
+        payGovUrl: vcapConstants.payGovClientUrl,
+        tcsAppID: tcsAppID
+      });
+    })
+    .catch(error => {
+      console.error(error);
+      return res.status(500).send();
+    });
+};
+
+const updatePermitWithError = (res, permit, paygovError) => {
+  updatePermit(permit, {
+    status: 'Error',
+    paygovError: JSON.stringify(paygovError)
+  }).then(updatedPermit => {
+    return getPermitError(res, updatedPermit);
+  });
+};
+
+const getPermitError = (res, permit) => {
+  let permitErrors = JSON.parse(permit.paygovError);
+  return res.status(400).json({
+    errors: [
+      {
+        status: 400,
+        errorCode: permitErrors.errorCode[0],
+        message: permitErrors.errorMessage[0],
+        permit: permitResult(permit, null)
+      }
+    ]
+  });
+};
+
+christmasTree.create = (req, res) => {
+  treesDb.christmasTreesForests
+    .findOne({
+      where: {
+        id: req.body.forestId
+      }
+    })
+    .then(forest => {
+      req.body.expDate = forest.endDate;
+      treesDb.christmasTreesPermits
+        .create(translatePermitFromClientToDatabase(req.body))
+        .then(permit => {
+          const xmlData = paygov.getXmlForToken(req.body.forestAbbr, req.body.orgStructureCode, permit);
+          postPayGov(xmlData)
+            .then(xmlResponse => {
+              xml2jsParse(xmlResponse, function(err, result) {
+                if (!err) {
+                  try {
+                    const token = paygov.getToken(result);
+                    return updatePermitWithToken(res, permit, token);
+                  } catch (error) {
+                    try {
+                      const paygovError = paygov.getResponseError(result);
+                      return updatePermitWithError(res, permit, paygovError);
+                    } catch (faultError) {
+                      throwError(faultError);
+                    }
+                  }
+                }
+              });
+            })
+            .catch(() => {
+              return res.status(500).send();
+            });
+        })
+        .catch(error => {
+          if (error.name === 'SequelizeValidationError') {
+            return res.status(400).json({
+              errors: error.errors
+            });
+          } else {
+            return res.status(500).send();
+          }
+        });
+    })
+    .catch(error => {
+      return res.status(400).json({
+        errors: error.errors
+      });
+    });
+};
+
+const sendEmail = (res, savedPermit, pngBuffer) => {
+  const attachments = [
+    {
+      filename: 'permit.png',
+      content: new Buffer(pngBuffer, 'utf-8'),
+      cid: 'unique@kreata.ee'
```

Comment:
what is this email?

Reply:
@lauraGgit it is an unique identifier of the attachment, we'll change this in sprint-7 story # 285